### PR TITLE
Update some array functions

### PR DESCRIPTION
**Pull request by Dorbedo**
* Usage of the new selectMin / selectMax commands
* ~~select update~~
* TODO for shuffle done

### CODE REVIEW

File: addons/arrays/fnc_findMin.sqf [hunk 1]
```diff
@@ -28,8 +28,5 @@ SCRIPT(findMin);
 
 if !(_array isEqualTypeAll 0) exitWith {nil};
 
-private _arraySorted = + _array;
-_arraySorted sort true; // true - ascending
-_arraySorted params ["_min"];
-
-[_min, _array find _min]
+private _max = selectMin _array;
+[_max, _array find _max]
```
Comment (commy2) [inline addons/arrays/fnc_findMin.sqf]: Variable name should still be `_min` :)

File: addons/arrays/fnc_shuffle.sqf [hunk 2]
```diff
@@ -14,38 +15,32 @@ Example:
...
-    [] + _this;
-} else {
-    if (IS_ARRAY(_this select 0)) then {
-        [] + (_this select 0); // Correct params passed.
-    } else {
-        WARNING("CBA_fnc_shuffle requires an array as first parameter, not just a direct array: " + str _this);
-        [] + _this;
-    };
-};
+_this params [["_array",[],[[]]],["_inPlace",false,[false]]];
```
Comment (commy2) [inline addons/arrays/fnc_shuffle.sqf]: The `_this` is superfluous as it's the variable used by the default unary syntax of `params`.

File: addons/arrays/fnc_select.sqf [hunk 3]
```diff
@@ -29,12 +29,4 @@ SCRIPT(select);
...
-private _result = [];
-
-{
-    if (_x call _filterCode) then {
-        _result pushBack _x;
-    };
-} forEach _array;
-
-_result
+_array select {_x call _filterCode}
```
Comment (commy2) [inline addons/arrays/fnc_select.sqf]: This makes the function no longer support `_forEachIndex`!

Comment (Dorbedo) [inline addons/arrays/fnc_select.sqf]: Ahhh, I remember. I'll remove.

File: addons/arrays/fnc_shuffle.sqf (hunk 2 above)
Comment (Dorbedo) [inline addons/arrays/fnc_shuffle.sqf]: Just personal style, the compiler is adding it anyways

I can remove it, if you like.

File: addons/arrays/fnc_findMin.sqf (hunk 1 above)
Comment (Dorbedo) [inline addons/arrays/fnc_findMin.sqf]: c&p, fixed

File: addons/arrays/fnc_shuffle.sqf (hunk 2 above)
Comment (commy2) [inline addons/arrays/fnc_shuffle.sqf]: The Arma compiler does not add `_this` to the unary syntax of `params`.

Comment (Dorbedo) [inline addons/arrays/fnc_shuffle.sqf]: You are right, I don't know why I was thinking this.

**Comment by commy2** [comment]
Re `CBA_fnc_select`. The same optimization was made to `CBA_fnc_reject` and no one complained.
https://github.com/Dorbedo/CBA_A3/commit/fc75b174e6fdc22ae4328940565451e26854622a

My thought was, that these functions are kinda deprecated now anyways so why risk breaking something that uses them?

**Comment by Dorbedo** [comment]
So mark them as deprecated and revert fc75b174e6fdc22ae4328940565451e26854622a too?

**Comment by commy2** [comment]
I'd say yes.

File: addons/arrays/fnc_shuffle.sqf [hunk 4]
```diff
@@ -14,38 +15,32 @@ Example:
...
-    _indexToRemove = floor random _size;
-    _shuffledArray pushBack (_tempArray deleteAt _indexToRemove);
-};
+private _tempArray =+ _array;
 
-_shuffledArray
+If (_inPlace) then {
+    for "_size" from (count _tempArray) to 1 step -1 do {
+        _array set [_size-1,(_tempArray deleteAt (floor random _size))];
+    };
```
Comment (PabstMirror) [inline addons/arrays/fnc_shuffle.sqf]: Should we still return the array here?

**Comment by commy2** [comment]
^ please change the tests in a separate pull request. This is to make sure that potential errors aren't missed.

**Comment by Dorbedo** [comment]
Alright.

**Comment by commy2** [comment]
```
 3:45:58 [CBA] (arrays) LOG: === Testing Arrays === x\cba\addons\arrays\test.sqf:13
 3:45:58 [CBA] (arrays) LOG: Testing CBA_fnc_filter x\cba\addons\arrays\test_filter.sqf:12
 3:45:58 [CBA] (arrays) Test OK: (CBA_fnc_filter is defined) x\cba\addons\arrays\test_filter.sqf:14
 3:45:58 [CBA] (arrays) Test OK: (str _result == str _expected) x\cba\addons\arrays\test_filter.sqf:20
 3:45:58 [CBA] (arrays) Test OK: (str _original == str _expected) x\cba\addons\arrays\test_filter.sqf:21
 3:45:58 [CBA] (arrays) Test OK: (str _result == str _expected) x\cba\addons\arrays\test_filter.sqf:26
 3:45:58 [CBA] (arrays) Test OK: (str _original != str _expected) x\cba\addons\arrays\test_filter.sqf:27
 3:45:58 [CBA] (arrays) Test OK: (str _result == str _expected) x\cba\addons\arrays\test_filter.sqf:32
 3:45:58 [CBA] (arrays) Test OK: (str _original == str _expected) x\cba\addons\arrays\test_filter.sqf:38
 3:45:58 [CBA] (arrays) Test OK: (str _result == str _expected) x\cba\addons\arrays\test_filter.sqf:39
 3:45:58 [CBA] (arrays) Test OK: (str _original == str _expected) x\cba\addons\arrays\test_filter.sqf:44
 3:45:58 [CBA] (arrays) Test OK: (str _result == str _expected) x\cba\addons\arrays\test_filter.sqf:45
 3:45:58 [CBA] (arrays) Test OK: (str _original == str _expected) x\cba\addons\arrays\test_filter.sqf:50
 3:45:58 [CBA] (arrays) Test OK: (str _result == str _expected) x\cba\addons\arrays\test_filter.sqf:51
 3:45:58 [CBA] (arrays) LOG: Testing CBA_fnc_inject x\cba\addons\arrays\test_inject.sqf:12
 3:45:58 [CBA] (arrays) Test OK: (CBA_fnc_inject is defined) x\cba\addons\arrays\test_inject.sqf:14
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_inject.sqf:18
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_inject.sqf:22
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_inject.sqf:26
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_inject.sqf:30
 3:45:58 [CBA] (arrays) LOG: Testing CBA_fnc_join x\cba\addons\arrays\test_join.sqf:12
 3:45:58 [CBA] (arrays) Test OK: (CBA_fnc_join is defined) x\cba\addons\arrays\test_join.sqf:14
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_join.sqf:18
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_join.sqf:22
 3:45:58 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_join.sqf:26
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_join.sqf:30
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_join.sqf:34
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_join.sqf:38
 3:45:59 File \x\cba\addons\arrays\fnc_shuffle.sqf [CBA_fnc_shuffle], line 1741
 3:45:59 [CBA] (arrays) ERROR: Test FAIL x\cba\addons\arrays\test_shuffle.sqf:27
 3:45:59             (count _result == count _original)
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_shuffle x\cba\addons\arrays\test_shuffle.sqf:12
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_shuffle is defined) x\cba\addons\arrays\test_shuffle.sqf:14
 3:45:59 [CBA] (arrays) Test OK: (count _result == count _original) x\cba\addons\arrays\test_shuffle.sqf:18
 3:45:59 [CBA] (arrays) Test OK: (_x in _original) x\cba\addons\arrays\test_shuffle.sqf:21
 3:45:59 [CBA] (arrays) Test OK: (_x in _original) x\cba\addons\arrays\test_shuffle.sqf:21
 3:45:59 [CBA] (arrays) Test OK: (_x in _original) x\cba\addons\arrays\test_shuffle.sqf:21
 3:45:59 [CBA] (arrays) Test OK: (count _result == count _original) x\cba\addons\arrays\test_shuffle.sqf:36
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_findNil x\cba\addons\arrays\test_findNil.sqf:13
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_findNil is defined) x\cba\addons\arrays\test_findNil.sqf:15
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNil.sqf:20
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNil.sqf:25
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNil.sqf:30
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNil.sqf:35
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNil.sqf:40
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_findNull x\cba\addons\arrays\test_findNull.sqf:13
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_findNull is defined) x\cba\addons\arrays\test_findNull.sqf:15
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNull.sqf:20
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNull.sqf:25
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNull.sqf:30
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNull.sqf:35
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findNull.sqf:40
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_findTypeName x\cba\addons\arrays\test_findTypeName.sqf:13
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_findTypeName is defined) x\cba\addons\arrays\test_findTypeName.sqf:15
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:22
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:26
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:30
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:34
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:38
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:42
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:46
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:50
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:54
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:58
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:62
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:66
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:70
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:74
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:78
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:82
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:87
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:92
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:97
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:102
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:107
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:112
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeName.sqf:117
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_findTypeOf x\cba\addons\arrays\test_findTypeOf.sqf:12
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_findTypeOf is defined) x\cba\addons\arrays\test_findTypeOf.sqf:14
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:19
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:24
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:29
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:34
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:39
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:44
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:49
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:54
 3:45:59 [CBA] (arrays) Test OK: (_result == _expected) x\cba\addons\arrays\test_findTypeOf.sqf:59
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_findMax x\cba\addons\arrays\test_findMax.sqf:13
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_findMax is defined) x\cba\addons\arrays\test_findMax.sqf:15
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMax.sqf:20
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMax.sqf:25
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMax.sqf:30
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMax.sqf:35
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMax.sqf:40
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMax.sqf:45
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMax.sqf:49
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMax.sqf:53
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMax.sqf:57
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_findMin x\cba\addons\arrays\test_findMin.sqf:12
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_findMin is defined) x\cba\addons\arrays\test_findMin.sqf:14
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMin.sqf:19
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMin.sqf:24
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMin.sqf:29
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMin.sqf:34
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMin.sqf:39
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_findMin.sqf:44
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMin.sqf:48
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMin.sqf:52
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMin.sqf:56
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMin.sqf:60
 3:45:59 [CBA] (arrays) Test OK: (isNil "_result") x\cba\addons\arrays\test_findMin.sqf:64
 3:45:59 [CBA] (arrays) LOG: Testing CBA_fnc_insert x\cba\addons\arrays\test_insert.sqf:12
 3:45:59 [CBA] (arrays) Test OK: (CBA_fnc_insert is defined) x\cba\addons\arrays\test_insert.sqf:14
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_insert.sqf:20
 3:45:59 [CBA] (arrays) Test OK: (_result isEqualTo _expected) x\cba\addons\arrays\test_insert.sqf:26
```

**Comment by commy2** [comment]
This pull request drops support for the deprecated syntax of the function:
```sqf
// Support the deprecated parameter style: [1, 2, 3, 4, 5] call CBA_fnc_shuffle.
_tempArray = if (count _this != 1) then {
    WARNING("CBA_fnc_shuffle requires an array as first parameter, not just a direct array: " + str _this);
    [] + _this;
} else {
    if (IS_ARRAY(_this select 0)) then {
        [] + (_this select 0); // Correct params passed.
    } else {
        WARNING("CBA_fnc_shuffle requires an array as first parameter, not just a direct array: " + str _this);
         [] + _this;
     };
 };
```

**Comment by commy2** [comment]
It's probably fine, because it's logged as deprecated anyways.

@Dorbedo can you please merge https://github.com/Dorbedo/CBA_A3/pull/1 ?

**Pull request merged**